test/hotspot/jtreg/serviceability/sa/ClhsdbSymbol.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) 2021, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import jdk.test.lib.apps.LingeredApp;
+import jtreg.SkippedException;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * @test
+ * @bug 8261095
+ * @summary Test the clhsdb 'symbol' command on live process
+ * @requires vm.hasSA
+ * @library /test/lib
+ * @run main/othervm ClhsdbSymbol
+ */
+
+public class ClhsdbSymbol {
+
+    public static void main(String[] args) throws Exception {
+        System.out.println("Starting the ClhsdbSymbol test");
+        LingeredApp theApp = null;
+
+        try {
+            List<String> cmds = null;
+            String cmdStr = null;
+            Map<String, List<String>> expStrMap = null;
+            ClhsdbLauncher test = new ClhsdbLauncher();
+            theApp = LingeredApp.startApp();
+            System.out.println("Started LingeredApp with pid " + theApp.getPid());
+
+            //Use command "class java.lang.Thread" to get the address of the InstanceKlass for java.lang.Thread
+            cmdStr = "class java.lang.Thread";
+            cmds = List.of(cmdStr);
+            expStrMap = new HashMap<>();
+            expStrMap.put(cmdStr, List.of("java/lang/Thread"));
+            String classOutput = test.run(theApp.getPid(), cmds, expStrMap, null);
+            String threadAddress = null;
+            String[] parts = classOutput.split("\n");
```

Comment:
I think you should be using the line.separator properly instead of "\n".
`            String linesep = System.getProperty("line.separator");`

Reply:
Yes. Coincidentally I was just having an internal chat about `split(linesep)`, and `String.lines()` was mentioned. We have a lot of existing code that uses `split(linesep)` because `String.lines()` has only been around since JDK 11, and even some newer test code we have uses `split(linsep)` because it was copied from older tests.